portfolio/src/main/java/com/google/sps/data/MergeBooks.java

```diff
@@ -0,0 +1,158 @@
+package com.google.sps.data;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.sps.data.Book;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+public class MergeBooks {
+  /**
+   * This method merges two books that have the same title
+   * and ISBN and returns a new book combined from both of them
+   * @param original: the original book
+   * @param addition: the book with additional info you
+   * want to add
+   * The order of the params does not matter
+   */
+  public static Book merge(Book original, Book addition) {
+    validate(original);
+    validate(addition);
+    String isbn = getNonNullString(original.isbn(),
```

Comment:
`Book` class is created using Autovalue, and it will never return a null value (unless explicitly specified, which we haven't). The default value that we are setting today is "N/A", so may be we should compare with that (or better use the empty string as default): https://screenshot.googleplex.com/T7M9563ixHg

This is applicable to other fields as well, please see how they should be handled.